### PR TITLE
chore(docs): Update links to official Shopify starter demo

### PR DESCRIPTION
**Pull request by jxnblk**
## Description

Updates links in Shopify-related docs to point to https://shopify-demo.gatsbyjs.com/

### Documentation

N/A

### CODE REVIEW

File: docs/docs/conceptual/gatsby-for-ecommerce.md [hunk 1]
```diff
@@ -38,10 +40,17 @@ E-commerce tends to have a number of specific requirements. When building a Gats
 - **Handling checkout.** In order to comply with PCI regulations around storing credit card information, e-commerce providers typically exert strong control over the "buy" or "checkout" experience. Shopify requires all checkout flows using their platform to use their hosted checkout pages, though it's common to wrap them in a subdomain of the main site (e.g. the Gatsby/Shopify site [strivectin.com](strivectin.com] wraps a `myshopify.com` URL under a `shop.strivectin.com` URL for checkout).
 - **Handling account pages.** Again, many sites choose to wrap their e-commerce provider’s account pages under their own domain.
 
-Additional resources:
+## Additional resources:
+
+<Announcement style={{ marginBottom: "1.5rem" }}>
```
Comment (jxnblk) [inline docs/docs/conceptual/gatsby-for-ecommerce.md:45]: I'm unsure whether or not we want to use this component for situations like this, and can remove this if not.

File: docs/docs/building-an-ecommerce-site-with-shopify.md [hunk 2]
```diff
@@ -4,7 +4,10 @@ title: Building an E-commerce site with Shopify
 
 In this tutorial, you will setup a new Gatsby website that fetches product data from [Shopify](https://www.shopify.com). The site displays a list of all products on a product listing page, and a page for every product in the store.
 
-If you are already comfortable with Gatsby and Shopify, you might want to check out the [Gatsby Shopify starter](https://www.gatsbyjs.com/starters/AlexanderProd/gatsby-shopify-starter/), which provides many of the same features as this example.
+If you're already comfortable with Gatsby and Shopify, you might want to check out our [Shopify Starter Demo][], a proof of concept showcasing 10,000 products and 30,000 SKUs (variants).
+You can clone the starter, host it on Gatsby and connect it to your own Shopify data to develop your own proof of concept in as little as an hour.
+
+[shopify starter demo]: https://shopify-demo.gatsbyjs.com/
```
Comment (meganesu) [inline docs/docs/building-an-ecommerce-site-with-shopify.md:10]: Looks like this link might be in the wrong place?

```suggestion
If you're already comfortable with Gatsby and Shopify, you might want to check out our [Shopify Starter Demo](https://shopify-demo.gatsbyjs.com/), a proof of concept showcasing 10,000 products and 30,000 SKUs (variants).
You can clone the starter, host it on Gatsby and connect it to your own Shopify data to develop your own proof of concept in as little as an hour.
```

**Review by meganesu — DISMISSED** [review]

**Review by thinkybeast — DISMISSED** [review]
👍 👍

File: docs/docs/building-an-ecommerce-site-with-shopify.md [hunk 3]
```diff
@@ -193,3 +194,8 @@ exports.createPages = async ({ graphql, actions }) => {
   })
 }
 ```
+
+## Additional Resources
+
+- [Gatsby Shopify Starter](https://shopify-demo.gatsbyjs.com/)
+- [Gatsby Shopify Hello World](/starters/ohduran/gatsby-starter-hello-world-shopify/)
```
Comment (meganesu) [inline docs/docs/building-an-ecommerce-site-with-shopify.md:201]: Do we still want to promote this other starter? If it's not up to date with the latest Shopify plugin I'd recommend we take it out. (We can still leave it in the starter library, but we don't need to promote it alongside the 30k one.)

Comment (jxnblk) [reply docs/docs/building-an-ecommerce-site-with-shopify.md:201]: I wasn't sure what to do with that one, but I'll take your recommendation here – makes a ton of sense! We can probably consider more resources if/when we pick up Shopify starter work again in the future
```suggestion
```

**Review by meganesu — APPROVED** [review]

**Pull request merged**